inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -167,11 +180,22 @@ ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(co
     DeviceMap<SoExecutableNetworkInternal> executableNetworkPerDevice;
     std::mutex load_mutex;
     std::vector<Task> loads;
+    std::once_flag readNetworkFlag;
     for (auto& p : metaDevices) {
         loads.push_back([&]() {
             const auto &deviceName = p.deviceName;
             const auto &deviceConfig = p.config;
-            auto exec_net = GetCore()->LoadNetwork(network, deviceName, deviceConfig);
+            SoExecutableNetworkInternal exec_net;
+            if (modelPath.empty()) {
+                exec_net = GetCore()->LoadNetwork(network, deviceName, deviceConfig);
+            } else if (GetCore()->DeviceSupportsImportExport(deviceName)) {
+                exec_net = GetCore()->LoadNetwork(modelPath, deviceName, deviceConfig);
```

Comment:
Checking that DeviceSupportsImportExport but then calling "Load" :)

Reply:
should I add some clarification comment?

Reply:
I would personally always call the LoadNetwork(modelPath) **without checking any support**, and if any exception happened would fallback to the conventional load.

Reply:
again, checking for Import is seemingly enough?

Reply:
The idea of this commit is to minimize number of 'ReadNetwork' calls, each call for usual network takes ~30-40ms

When device supports caching - it will not 'ReadNetwork' in case of cache available
When device doesn't support it - LoadNetwork(modelPath) will not throw, it will ReadNetwork(modelPath)+LoadNetwork(readCnnNet).
In this case, when doing the same 'ReadNetwork' for all devices - it will be not efficient for load time